bika/lims/upgrade/v01_03_000.py

```diff
@@ -1198,3 +1225,182 @@ def decouple_analysis_requests_from_cancellation_workflow(portal):
+    del_index(portal, CATALOG_ANALYSIS_REQUEST_LISTING, "getSampleID")
+    del_index(portal, CATALOG_ANALYSIS_REQUEST_LISTING, "getSampleUID")
+    del_metadata(portal, "bika_catalog", "getSampleID")
+    del_metadata(portal, CATALOG_ANALYSIS_LISTING, "getExpiryDate")
+    del_metadata(portal, CATALOG_ANALYSIS_LISTING, "getSamplePartitionID")
+    del_metadata(portal, CATALOG_ANALYSIS_REQUEST_LISTING, "getSampleID")
+    del_metadata(portal, CATALOG_ANALYSIS_REQUEST_LISTING, "getSampleUID")
+    del_metadata(portal, CATALOG_ANALYSIS_REQUEST_LISTING, "getSampleURL")
+
+
+def fix_analysisrequests_in_sampled_status(portal):
+    logger.info("Resolving status 'sample' from Analysis Requests ...")
+    wf_id = "bika_ar_workflow"
+    wf_tool = api.get_tool("portal_workflow")
+    workflow = wf_tool.getWorkflowById(wf_id)
+    query = dict(portal_type="AnalysisRequest", review_state="sampled")
+    brains = api.search(query, CATALOG_ANALYSIS_REQUEST_LISTING)
+    total = len(brains)
+    for num, brain in enumerate(brains):
+        if num % 100 == 0:
+            logger.info("Resolving state to 'sample_due': {}/{}"
+                        .format(num, total))
+        analysis_request = api.get_object(brain)
+        changeWorkflowState(analysis_request, wf_id, "sample_due")
+        workflow.updateRoleMappingsFor(analysis_request)
+        analysis_request.reindexObject(idxs=["review_state"])
+
+
+def port_analysis_request_proxy_fields(portal):
+    logger.info("Purging Analysis Request Proxy Fields ...")
+
+    def set_value(analysis_request, field_name, field_value):
+        ar_field = analysis_request.Schema()[field_name]
+        if ar_field.type == 'uidreference':
+            if api.is_object(field_value):
+                field_value = api.get_uid(field_value)
+            elif not api.is_uid(field_value):
+                return
+        ar_field.set(analysis_request, field_value)
+
+    def unlink_proxy_fields(sample_brain, analysis_request=None):
+        processed_fields = []
+        sample_obj = api.get_object(sample_brain)
+        if not analysis_request:
+            for ar in sample_obj.getAnalysisRequests():
+                processed_fields.extend(unlink_proxy_fields(sample_brain, ar))
+                ar.reindexObject()
+            return list(set(processed_fields))
+
+        for field_id in PROXY_FIELDS_TO_PURGE:
+            field_value = None
+            try:
+                field = sample_obj.Schema().getField(field_id)
+                field_value = field.get(sample_obj)
+            except AttributeError:
+                logger.warn("Field {} not found for {}"
+                            .format(field_id, sample_obj.getId()))
+            if not field_value:
+                continue
+            set_value(analysis_request, field_id, field_value)
+            processed_fields.append(field_id)
+        return processed_fields
+
+    start = time.time()
+    query = dict(portal_type="Sample")
```

Comment:
Yes, it should. Since I was removing Proxy fields little by little I wanted the upgrade step to always process all Samples to be sure that all was working properly... at the end, I missed to add the `isValid` filter, as well as to do `sample.setMigrated(True)`